packages/commonwealth/client/scripts/views/components/SublayoutHeader/useUserMenuItems.tsx

```diff
@@ -122,20 +123,52 @@ const useUserMenuItems = ({
     setSelectedAddress,
   ]);
 
-  const addresses: PopoverMenuItem[] = userData.accounts.map((account) => {
-    const communityCaip2Prefix = chainBaseToCaip2(account.community.base);
-    const communityIdOrPrefix =
-      account.community.base === ChainBase.CosmosSDK
-        ? account.community.ChainNode?.bech32
-        : account.community.ChainNode?.ethChainId;
-    const communityCanvasChainId = chainBaseToCanvasChainId(
-      account.community.base,
-      // @ts-expect-error StrictNullChecks
-      communityIdOrPrefix,
+  const [canvasSignedAddresses, setCanvasSignedAddresses] = useState<string[]>(
+    [],
+  );
+
+  const updateCanvasSignedAddresses = useCallback(async () => {
+    const signedAddresses: string[] = [];
+
+    await Promise.all(
+      userData.accounts.map(async (account) => {
+        // making a fresh query to get chain and community info for this address
+        // as all the necessary fields don't exist on user.address, these should come
+        // from api in the user address response, and the extra api call here removed
+        const community = await EXCEPTION_CASE_VANILLA_getCommunityById(
```

Comment:
same, in hook, can't we use useGetCommunityByIdQuery?

Reply:
answered [here](https://github.com/hicommonwealth/commonwealth/pull/8877#discussion_r1723621623)